corehq/apps/app_manager/static/app_manager/js/report-module.js

```diff
@@ -1,17 +1,108 @@
 
 var ReportModule = (function () {
 
-    function ReportConfig(report_id, display, availableReportIds, language) {
+    function Config(dict) {
+        var self = this;
+
+        dict = dict || {};
+
+        self.keyValuePairs = ko.observableArray();
+        for (var key in dict) {
+            self.keyValuePairs.push([ko.observable(key), ko.observable(dict[key])]);
+        }
+
+        self.addConfig = function() {
+            self.keyValuePairs.push([ko.observable(''), ko.observable('')]);
```

Comment:
this is probably knockout 101 question, but why do you need an entry with blank key and blank value?